### PR TITLE
feat(InlineEditing): add placeholder support

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

We can't specify placeholders

**What is the chosen solution to this problem?**

Add placeholder support

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/design-system/src/components/InlineEditing/InlineEditing.style.ts [hunk 1]
```diff
@@ -38,6 +38,11 @@ export const InlineEditing = styled.div`
 			overflow: hidden;
 			white-space: nowrap;
 			flex: 0 1 auto;
+
+			&.placeholder[data-text]::before {
```
Comment (frassinier) [inline packages/design-system/src/components/InlineEditing/InlineEditing.style.ts:42]:
```suggestion
			&[data-placeholder]::before {
```

File: packages/design-system/src/components/InlineEditing/InlineEditing.tsx [hunk 2]
```diff
@@ -133,19 +136,22 @@ const InlineEditing = React.forwardRef(
 						onDoubleClick={loading ? undefined : () => setEditMode(true)}
 					>
 						<S.InlineEditingValue
-							className="c-inline-editing__value"
+							className={classNames('c-inline-editing__value', { placeholder: !value })}
```
Comment (frassinier) [inline packages/design-system/src/components/InlineEditing/InlineEditing.tsx:139]:
```suggestion
							className="c-inline-editing__value"
```

File: packages/design-system/src/components/InlineEditing/InlineEditing.tsx [hunk 3]
```diff
@@ -133,19 +136,22 @@ const InlineEditing = React.forwardRef(
...
+							data-text={placeholder}
```
Comment (frassinier) [inline packages/design-system/src/components/InlineEditing/InlineEditing.tsx:140]:
```suggestion
							data-placeholder={placeholder}
```

File: packages/design-system/src/components/InlineEditing/InlineEditing.tsx [hunk 4]
```diff
@@ -133,19 +136,22 @@ const InlineEditing = React.forwardRef(
...
 							as={renderValueAs || renderAs}
+							aria-hidden
```
Comment (frassinier) [inline packages/design-system/src/components/InlineEditing/InlineEditing.tsx:142]: Should be aria-hidden *only* if there is no value
```suggestion
							aria-hidden={value !== undefined}
```

File: packages/design-system/src/components/InlineEditing/InlineEditing.tsx [hunk 5]
```diff
@@ -47,6 +47,8 @@ const InlineEditing = React.forwardRef(
 			hasError,
 			required,
 			renderAs,
+			ariaLabel,
```
Comment (frassinier) [inline packages/design-system/src/components/InlineEditing/InlineEditing.tsx:50]: Not defined into `InlineEditingProps`,  it should be mandatory

**Review by frassinier — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Pull request merged**